stan/math/prim/mat/fun/ViennaCL.hpp

```diff
@@ -0,0 +1,33 @@
+#ifndef STAN_MATH_PRIM_MAT_FUN_VIENNACL_HPP
+  #define STAN_MATH_PRIM_MAT_FUN_VIENNACL_HPP
+
+  #ifdef STAN_GPU
+    #ifndef VIENNACL_WITH_OPENCL
+      #define VIENNACL_WITH_OPENCL 1
+    #endif
+  #endif
+
+  #ifndef VIENNACL_WITH_EIGEN
+    #define VIENNACL_WITH_EIGEN 1
+  #endif
+
+  #include <viennacl/vector.hpp>
+  #include <viennacl/matrix.hpp>
+  #include <viennacl/linalg/direct_solve.hpp>
+  #include <viennacl/linalg/lu.hpp>
+  #include <viennacl/linalg/ilu_operations.hpp>
+
+  #ifdef STAN_GPU
+    std::ifstream in("stan/math/prim/mat/fun/custom_kernels.cl", std::ios::in | std::ios::binary);
```

Comment:
are all these things dropped into the global namespace? Perhaps we can wrap them inside `stan::math` unless there's some reason they need to be global.